File: app/schemas/history.py
"""User history response schemas."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class CareerHistoryItem(BaseModel):
    id: int
    language: str
    interests: list[str]
    result: dict[str, Any]
    created_at: datetime


class LearningPathHistoryItem(BaseModel):
    id: int
    career_name: str
    path: dict[str, Any]
    progress: dict[str, bool]
    is_active: bool
    created_at: datetime
    updated_at: datetime


class UserHistoryResponse(BaseModel):
    assessments: list[CareerHistoryItem]
    learning_paths: list[LearningPathHistoryItem]

File: src/api/history.ts
import apiClient from './client';
import { CareerRecommendResponse, LearningPath } from '../types';

export interface UserHistory {
  assessments: {
    id: number;
    language: string;
    interests: string[];
    result: CareerRecommendResponse;
    created_at: string;
  }[];
  learning_paths: {
    id: number;
    career_name: string;
    path: LearningPath;
    progress: Record<string, boolean>;
    is_active: boolean;
    created_at: string;
    updated_at: string;
  }[];
}

export const historyApi = {
  getAll() {
    return apiClient.get<UserHistory>('/history');
  },
};
